src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java

```diff
@@ -19,89 +20,111 @@
         Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
         cdstP = cdst.clone();
         Mat linesP = new Mat();
         Imgproc.HoughLinesP(dst, linesP, 3, Math.PI, 200, RectangleDetector.VERTICAL_LINE_LENGTH, 1);
-        log.debug("Quantity lines " + linesP.size().toString());
+        log.debug("Quantity of lines founded =  " + linesP.size().toString());
         List<double[]> lines = new ArrayList<>();
         for (int k = 0; k < linesP.rows(); k++) {
             lines.add(linesP.get(k, 0));
         }
         if (lines.size() > MINIMAL_QUANTITY_LINES_FOR_TABLE) {
             log.debug("Result: table found!");
-            log.debug("Is table exist on page finished.");
+            log.debug("Method IsTableExistOnPage finished.");
             return true;
         } else {
             log.debug("Result: table not found");
-            log.debug("Is table exist on page finished.");
+            log.debug("Method IsTableExistOnPage finished.");
             return false;
         }
-
     }
 
-    public String detectTable(String fileSource) {
-        log.debug("Class TableRecognizer.detectTable started.");
+    public String detectTable(String fileSource, int page) {
+        log.debug("Method detectTable started.");
         OpenCV.loadLocally();
         Mat dst = new Mat(), cdst = new Mat(), cdstP, cropTable = new Mat();
         Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
         Imgproc.Canny(source, dst, 50, 200, 3, false);
-        Imgcodecs.imwrite("afterCanny.png", dst);
+        Imgcodecs.imwrite(page + "_#7_afterCanny.png", dst);
         Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
         cdstP = cdst.clone();
         Mat linesP = new Mat();
         Imgproc.HoughLinesP(dst, linesP, 1, Math.PI / 180, 25, 700, 10);
+        double[] array = linesP.get(0, 0);
+        x1 = array[0];
+        y1 = array[1];
+        x2 = array[2];
+        y2 = array[3];
+        y3 = array[3];
+
         for (int x = 0; x < linesP.rows(); x++) {
             double[] l = linesP.get(x, 0);
             Imgproc.line(cdstP, new Point(l[0], l[1]), new Point(l[2], l[3]), new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0);
-            if (l[1] > y1) {
-                y1 = l[1];
-                x2 = l[2];
-                y2 = l[3];
+            if (l[0] < x1) {
+                x1 = l[0];
             }
-            if (l[1] < y3) {
-                y3 = l[1];
+            if (l[1] < y1) {
+                y1 = l[1];
             }
             if (l[2] > x2) {
                 x2 = l[2];
             }
-            if (l[0] < x1) {
-                x1 = l[0];
+            if (l[3] > y3) {
+                y3 = l[3];
             }
+
         }
-        Imgcodecs.imwrite("afterHough.png", cdstP);
-        Rect rectangle = createRect(x1, y1, x2, y2, y3);
+        Imgcodecs.imwrite(page + "_#8_afterHough.png", cdstP);
+        log.debug("x1 = {}; y1 = {}; x2 = {}; y2={}, y3={}", x1, y1, x2, y2, y3);
```

Comment:
INfo level

Reply:
I think it's debug level.
INFO
INFO messages are like the normal behavior of applications. They state what happened. For example, if a particular service stopped or started or you added something to the database. These entries are nothing to worry about during usual operations. The information logged using the INFO log is usually informative, and it does not necessarily require you to follow up on it.

DEBUG
With DEBUG, you are giving diagnostic information in a detailed manner. It is verbose and has more information than you would need when using the application. DEBUG logging level is used to fetch information needed to diagnose, troubleshoot, or test an application. This ensures a smooth running application.